code/modules/mob/living/ventcrawling.dm

```diff
@@ -1,12 +1,75 @@
-// VENTCRAWLING
-// Handles the entrance and exit on ventcrawling
+#define VENT_SOUND_DELAY 30
+
+/obj/effect/ventcrawl_holder
+	name = "the pipe"//should be set to the name of the pipe we should be in
+	density = FALSE
+	anchored = FALSE
+
+	///what machine we're currently in
+	var/obj/machinery/atmospherics/current_node
+
+/obj/effect/ventcrawl_holder/proc/setup_on_node(mob/living/package, obj/machinery/atmospherics/starting_node)
+	if(!starting_node?.associated_loc || !package)
+		return FALSE
+
+	package.client?.eye = src
+	package.forceMove(src)
+	current_node = starting_node
+	package.ventcrawl_layer = starting_node.piping_layer
+	RegisterSignal(package, list(COMSIG_MOVABLE_MOVED, COMSIG_PARENT_PREQDELETED), .proc/check_package)
```

Comment:
PRE qdeleted?

Reply:
yes?

Reply:
Well for one it's called before gc_destroyed is set, so your QDELETED check won't work
And for two it's the "do you want to cancel this qdel" signal, not the "react to this qdel" signal, so if something uses it to cancel the operation you're gonna be shit out of luck 
![image](https://user-images.githubusercontent.com/58055496/149857747-f3d103cd-ba86-4bd6-9f43-48cda6cba923.png)

Also why register both these signals to check_package if you're going to need to check for both individually? in fact, why do you need to listen for destroy at all? won't a move to nullspace cause that to happen regardless?

Reply:
good point